test/appservice/utils/webAPIClient.ts

```diff
@@ -0,0 +1,54 @@
+import * as request from "request";
+import { MatrixClient } from "matrix-bot-sdk";
+
+interface OpenIDTokenInfo {
+    access_token: string,
+    expires_in: number,
+    matrix_server_name: string,
+    token_type: string
+}
+
+async function getOpenIDToken(client: MatrixClient): Promise<string> {
+    const tokenInfo: OpenIDTokenInfo = await client.doRequest("POST", `/_matrix/client/v3/user/${await client.getUserId()}/openid/request_token`, undefined, {});
+    return tokenInfo.access_token;
+}
+
+export interface CreateMjolnirResponse {
+    mjolnirUserId: string,
+    managementRoomId: string,
+}
+
+export class MjolnirWebAPIClient {
+
+    private constructor(
+        private readonly matrixClient: MatrixClient,
+        /**
+         * This will want refactoring later on since, well, for some reason our API is designed to accept the token in request body and not authorization header :( 
```

Comment:
Certainly true, but that's not documentation :)

Is this something that can be done quickly?

Reply:
Probably not https://github.com/matrix-org/mjolnir/issues/416